pkg/descheduler/strategies/highnodeutilization.go

```diff
@@ -0,0 +1,205 @@
+func classifyNodesForHighUtilization(npm NodePodsMap, thresholds api.ResourceThresholds, targetThresholds api.ResourceThresholds) ([]NodeUsageMap, []NodeUsageMap) {
+	desiredNodes, targetNodes := []NodeUsageMap{}, []NodeUsageMap{}
+	for node, pods := range npm {
+		usage := nodeUtilization(node, pods)
+		nuMap := NodeUsageMap{
+			node:    node,
+			usage:   usage,
+			allPods: pods,
+		}
+		if isNodeBelowThresholdUtilization(usage, targetThresholds) {
+			klog.V(2).Infof("Node %#v is under utilized with usage: %#v", node.Name, usage)
+			targetNodes = append(targetNodes, nuMap)
+		} else if !nodeutil.IsNodeUnschedulable(node) &&
+			!isNodeAboveThresholdUtilization(usage, thresholds) &&
+			!isNodeBelowThresholdUtilization(usage, targetThresholds) {
+			klog.V(2).Infof("Node %#v is utilized with usage: %#v", node.Name, usage)
+			desiredNodes = append(desiredNodes, nuMap)
+		} else {
+			klog.V(2).Infof("Node %#v is over utilized with usage: %#v", node.Name, usage)
+		}
+
+	}
+	return desiredNodes, targetNodes
+}
+
+// evictPodsFromTargetNodes evicts pods based on priority, if all the pods on the node have priority, if not
+// evicts them based on QoS as fallback option.
+func evictPodsFromHighUsageTargetNodes(
+	ctx context.Context,
+	targetNodes, desiredNodes []NodeUsageMap,
+	thresholds api.ResourceThresholds,
+	podEvictor *evictions.PodEvictor,
+) {
+
+	sortNodesByUsage(targetNodes, false)
+
+	// upper bound on total number of pods/cpu/memory to be moved
+	totalPods, totalCPU, totalMem, taintsOfDesiredNodes := computeDesiredNodeResourcesAndTaints(desiredNodes, thresholds)
+
+	klog.V(1).Infof("Total capacity to be moved: CPU:%v, Mem:%v, Pods:%v", totalCPU, totalMem, totalPods)
+	klog.V(1).Infof("********Number of pods evicted from each node:***********")
+
+	for _, node := range targetNodes {
+		klog.V(3).Infof("evicting pods from node %#v with usage: %#v", node.node.Name, node.usage)
+
+		nonRemovablePods, removablePods := classifyPods(node.allPods, podEvictor)
+		klog.V(2).Infof("allPods:%v, nonRemovablePods:%v, removablePods:%v", len(node.allPods), len(nonRemovablePods), len(removablePods))
+
+		if len(removablePods) == 0 {
+			klog.V(1).Infof("no removable pods on node %#v, try next node", node.node.Name)
+			continue
+		}
+
+		klog.V(1).Infof("evicting pods based on priority, if they have same priority, they'll be evicted based on QoS tiers")
+		// sort the evictable Pods based on priority. This also sorts them based on QoS. If there are multiple pods with same priority, they are sorted based on QoS tiers.
+		podutil.SortPodsBasedOnPriorityLowToHigh(removablePods)
+		evictPodsFromHighUsageNode(ctx, removablePods, &totalPods, &totalCPU, &totalMem, taintsOfDesiredNodes, podEvictor, node.node)
+
+		klog.V(1).Infof("%v pods evicted from node %#v with usage %v", podEvictor.NodeEvicted(node.node), node.node.Name, node.usage)
+	}
+}
+
+func evictPodsFromHighUsageNode(
```

Comment:
`evictPods` and `evictPodsFromHighUsageNode` should be quite similar. Or, identical. At the end it's just about selecting a list of nodes from which a set of pods get evicted while still checking sufficient amount of resources is still available.